v1/authorization/caching_password_service.go

```diff
@@ -0,0 +1,110 @@
+package authorization
+
+import (
+	"bytes"
+	"context"
+	crand "crypto/rand"
+	"crypto/sha256"
+	"io"
+	"sync"
+
+	"github.com/influxdata/influxdb/v2"
+)
+
+// An implementation of influxdb.PasswordsService that will perform
+// ComparePassword requests at a reduced cost under certain
+// conditions. See ComparePassword for further information.
+//
+// The cache is only valid for the duration of the process.
+type CachingPasswordsService struct {
+	inner influxdb.PasswordsService
+
+	mu        sync.RWMutex // protects concurrent access to authCache
+	authCache map[influxdb.ID]authUser
```

Comment:
What are the bounds on this in memory cache.. or do we not need to worry about it growing unbounded?

Reply:
Good question – this is only for unique user name tokens, so we'd expect this to be probably dozens at most